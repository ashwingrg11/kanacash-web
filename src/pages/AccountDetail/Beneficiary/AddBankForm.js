import React, { Component } from "react";
import { Container, Row, Col } from "react-bootstrap";
import ROUTES from "../../../assets/Routes/Routes";
import MyAccountSidebar from "../../../components/MyAccountSidebar";
import { withRouter } from "react-router-dom";
import Header from "../../../containers/layout/Header";
import { getBeneficiaries } from "../../../store/actions/senderDetail";
import { connect } from "react-redux";
import Footer from "../../../containers/layout/Footer";

import AccountHeader from "../../../containers/user/AccountHeader";
import BankForm from "../../../containers/user/BankForm";
import * as api from "../../../services/axios/auxApi";
import PropTypes from "prop-types";
import Disclaimer from "../../../components/Disclaimer/Disclaimer";

class AddBankForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      beneficiaryList: [],
      activePage: "",
    };
  }

  addBank = (values) => {
    let body = {
      beneficiaryId: this.props.location.state.beneficiaryReferenceId,
      bank_id: parseInt(values.bank),
      account_number: JSON.stringify(values.accountNumber),
    };
    api
      .createBeneficiariesBank(body)
      .then((res) => {
        this.props.history.push({
          pathname: `${ROUTES.BENEFICIARY_LIST}`,
          state: {
            showAlert: true,
            message: "New beneficiary information has been added successfully.",
            variant: "success",
          },
        });
      })
      .catch((err) => {
        // console.log("error while creating beneficiary bank", err);
      });
  };

  render() {
    return (
      <>
        <Header />
        <div className="shadow-line"></div>
        <section className="admin-wrapper">
          <Container>
            <Row>
              <Col>
                <AccountHeader />
                <Row className="admin-content">
                  <Col xl="3" lg="4" md="5" sm="12">
                    <MyAccountSidebar />
                  </Col>
                  <Col xl="9" lg="8" md="7" sm="12">
                    <Row>
                      <Col className="activity-title mb-2">New Beneficiary</Col>
                    </Row>
                    <Row>
                      <Col>
                        <div className="payment-block-wrap">
                          <h3 className="subhead-title">
                            Add beneficiary bank
                          </h3>
                          <Row>
                            <Col className="theme-form">
                              <BankForm
                                handleChange={this.addBank}
                                country={this.props?.location?.state?.country}
                                handleBack={() =>
                                  this.props.history.push(
                                    `${ROUTES.BENEFICIARY_LIST}`
                                  )
                                }
                              />
                            </Col>
                          </Row>
                        </div>
                      </Col>
                    </Row>
                  </Col>
                </Row>
              </Col>
            </Row>
          </Container>
        </section>
        <Footer />
        {!localStorage.acceptPolicy && <Disclaimer />}
      </>
    );
  }
}
const mapStateToProps = (state) => ({
  beneficiary: state.senderDetail.beneficiaries,
  activePage: state.senderDetail.activePage,
  acceptPolicy: state.auth.accept_policy,
});
const mapActionToProps = { getBeneficiaries };

AddBankForm.propTypes = {
  location: PropTypes.object,
  history: PropTypes.object,
};

export default connect(
  mapStateToProps,
  mapActionToProps
)(withRouter(AddBankForm));
