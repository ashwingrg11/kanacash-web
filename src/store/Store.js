import { createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { composeWithDevTools } from 'redux-devtools-extension';
import Thunk from "redux-thunk";
import LoginReducer from "./reducers/Login";
import ForgotPasswordReducer from "./reducers/ForgotPassword";
import LoaderReducer from "./reducers/Loader";
import ModalReducer from "./reducers/Modal";
import NetworkReducer from "./reducers/Network";
import AuthReducer from "./reducers/Auth";
import ErrorReducer from "./reducers/Error";
import MiscellaneousReducer from "./reducers/MiscellaneousReducer";
import senderDetailReducers from "./reducers/SenderDetailReducer";
import TransactionReducer from "./reducers/TransactionReducer";
import SignUpReducer from "./reducers/SignUpReducer";
import AdminReducers from "./reducers/adminReducers";

const middleware = [Thunk];
const initialState = {};

const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
      })
    : compose;

const rootReducer = combineReducers({
  login: LoginReducer,
  forgotPassword: ForgotPasswordReducer,
  loader: LoaderReducer,
  modal: ModalReducer,
  network: NetworkReducer,
  auth: AuthReducer,
  error: ErrorReducer,
  miscellaneous: MiscellaneousReducer,
  senderDetail: senderDetailReducers,
  transaction: TransactionReducer,
  signup: SignUpReducer,
  admin: AdminReducers,
});

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);
export default store;
