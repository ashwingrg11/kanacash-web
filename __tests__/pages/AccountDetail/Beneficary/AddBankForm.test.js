import React from "react";
import AddBankForm from "../../../../src/pages/AccountDetail/Beneficiary/AddBankForm";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import store from "../../../../src/store/Store";

import renderer from "react-test-renderer";
describe("<AddBankForm />", () => {

  it("AddBankForm renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <AddBankForm/>
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
