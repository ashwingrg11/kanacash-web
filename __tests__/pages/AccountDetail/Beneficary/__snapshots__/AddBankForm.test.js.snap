// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBankForm /> AddBankForm renders correctly 1`] = `
Array [
  <header
    className="header-main"
  >
    <div
      className="container"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          className="navbar-brand"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <img
            alt="logo"
            className="logo"
            src=""
          />
        </div>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="ml-md-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <span
              className="btn btn-primary btn-theme btn-primary-border"
              onClick={[Function]}
            >
              Sign in
            </span>
            <button
              className="btn btn-primary btn-theme btn-primary-theme"
              onClick={[Function]}
            >
              Sign up
            </button>
          </div>
        </div>
      </nav>
    </div>
  </header>,
  <div
    className="shadow-line"
  />,
  <section
    className="admin-wrapper"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h2
            className="account-title"
          >
            My Account
          </h2>
          <p>
            Everything we know about you.
          </p>
          <div
            className="admin-content row"
          >
            <div
              className="col-xl-3 col-lg-4 col-md-5 col-sm-12"
            >
              <ul
                className="sidebar-menu"
              >
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#1077BE",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="activeImg"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Profile
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Beneficiary
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Bank Accounts
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Debit Cards
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Transactions
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Transaction Limit
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Settings
                  </div>
                </li>
              </ul>
            </div>
            <div
              className="col-xl-9 col-lg-8 col-md-7 col-sm-12"
            >
              <div
                className="row"
              >
                <div
                  className="activity-title mb-2 col"
                >
                  New Beneficiary
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <div
                    className="payment-block-wrap"
                  >
                    <h3
                      className="subhead-title"
                    >
                      Add beneficiary bank
                    </h3>
                    <div
                      className="row"
                    >
                      <div
                        className="theme-form col"
                      >
                        <form
                          className="theme-form"
                          noValidate={true}
                          onSubmit={[Function]}
                        >
                          <div
                            className="form-row"
                          >
                            <div
                              className="form-group col"
                            >
                              <label>
                                Pleae select bank
                              </label>
                              <select
                                className="form-control"
                                name="bank"
                                onChange={[Function]}
                                value=""
                              >
                                <option
                                  value=""
                                >
                                  Select the bank
                                </option>
                                <option
                                  value=""
                                >
                                  LOADING...
                                </option>
                              </select>
                              <div
                                className="invalid-feedback"
                              />
                            </div>
                          </div>
                          <div
                            className="form-row"
                          >
                            <div
                              className="form-group col"
                            >
                              <label>
                                Account number
                              </label>
                              <input
                                className="special-input form-control"
                                name="accountNumber"
                                onChange={[Function]}
                                placeholder="Add account number"
                                type="number"
                                value=""
                              />
                              <div
                                className="invalid-feedback"
                              />
                            </div>
                          </div>
                          <div
                            className="button-group"
                          >
                            <button
                              className="bt btn-secondary btn-secondary-big btn btn-primary"
                              disabled={false}
                              type="submit"
                            >
                              Save
                            </button>
                            <button
                              className="btn btn-back-arrow float-right"
                              onClick={[Function]}
                            >
                              <img
                                alt="BackArrow"
                                src=""
                              />
                              Back
                               
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer
    className="footer-section text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="foot-nav"
          >
            <a
              aria-current="page"
              className="link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              aria-current={null}
              className="link"
              href="/about-us"
              onClick={[Function]}
            >
              About Us
            </a>
            <a
              aria-current={null}
              className="link"
              href="/terms-of-service"
              onClick={[Function]}
            >
              Terms
            </a>
            <a
              aria-current={null}
              className="link"
              href="/privacy-policy"
              onClick={[Function]}
            >
              Privacy Policy
            </a>
            <a
              aria-current={null}
              className="link"
              href="/copyright"
              onClick={[Function]}
            >
              Copyright
            </a>
            <a
              aria-current={null}
              className="link"
              href="/contact-us"
              onClick={[Function]}
            >
              Contact
            </a>
          </div>
        </div>
        <div
          className="copyright col-sm-12"
        >
          <p>
            © Golden Money Transfer Inc. 2020. All Rights Reserved.
             
            <b>
              Licensed by
            </b>
             Department of Financial Institutions, Nevada
          </p>
        </div>
      </div>
    </div>
  </footer>,
  <div
    className="disclaimer text-center"
  >
    <div
      className="container"
    >
      <div
        className="policy"
      >
         
        This website uses cookies to ensure you get the best experience on our website, as described in our
         
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            privacy policy.
        </span>
      </div>
      <div
        className="policy1"
      >
        This website uses cookies,
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            learn more.
        </span>
      </div>
      <div
        className="buttonDiv"
      >
        <button
          className="btn-outline-warning"
          onClick={[Function]}
          style={
            Object {
              "border": "none",
              "borderRadius": "50px",
              "color": "black",
              "padding": "5px 15px",
            }
          }
        >
          Accept
        </button>
      </div>
    </div>
  </div>,
]
`;
